core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -74,13 +74,12 @@ class SparkHadoopUtil extends Logging {
     }
   }
 
-  /**
-   * Return an appropriate (subclass) of Configuration. Creating config can initializes some Hadoop
-   * subsystems.
-   */
-  def newConfiguration(conf: SparkConf): Configuration = {
-    val hadoopConf = new Configuration()
 
+  /**
+    * Appends S3-specific, spark.hadoop.*, and spark.spark.buffer.size configurations to a Hadoop
```

Comment:
That is a very old TODO, but if I remember correctly, it was a result of me omitting code as I copied logic from Shark.  It would be good to understand what the job that was tested on a cluster is doing, and why it needs info in the hive conf.  Just because long term we are moving further and further away from using hive code.

No objections though if this is fixing something.